config/configtls/configtls.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configtls
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+)
+
+// TLSConfig exposes the common client and server TLS configurations.
+// Note: Since there isn't anything specific to a server connection. Components
+// with server connections should use TLSConfig.
+type TLSConfig struct {
```

Comment:
s/TLSConfig/TLSSettings? See motivation in the next comment.

Reply:
Agreed that makes more sense. Will update.